nautobot/extras/api/serializers.py

```diff
@@ -343,6 +344,20 @@ class ConfigContextSerializer(ValidatedModelSerializer, NotesSerializerMixin):
     )
     tags = serializers.SlugRelatedField(queryset=Tag.objects.all(), slug_field="slug", required=False, many=True)
 
+    dynamic_groups = SerializedPKRelatedField(
+        queryset=DynamicGroup.objects.all(),
+        serializer=NestedDynamicGroupSerializer,
+        required=False,
+        many=True,
+    )
+
+    # Conditional enablement of dynamic groups filtering
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not settings.CONFIG_CONTEXT_DYNAMIC_GROUPS_ENABLED:
+            self.fields.pop("dynamic_groups")
```

Comment:
Relating to #3368 as well, we're having some discussions about how we should handle variable REST APIs whose schema depends on things like system configuration, and how that can be accommodated by SDKs like `pynautobot` and `nautobot-ansible`. Depending on how that discussion turns out, it *might* be better for `dynamic_groups` to always be present in the schema (but just be discarded/rejected on input if the feature is disabled) instead of having the field be conditional in its existence.

Reply:
We discussed this further and the point was made that the schema is already not a constant static thing. This adds another way in which that is the case, but IMO it's not making the problem necessarily any worse, so I'll withdraw my hold.

Reply:
Thanks for the info!